### PR TITLE
Fix typos

### PR DESCRIPTION
**Pull request by larrybradley**
Note that this includes a name change of parameter/property from `indicies` to `indices`.  Formally this is a breaking change, but I did not see any usage of `DataSelectedMessage` in the code.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [ ] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: docs/dev/ui_description.rst [hunk 1]
```diff
@@ -268,7 +268,7 @@ implemented inside a widget:
 Glue callback properties
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
-In cases where trailets are not apppropriate (e.g. where some python object is
+In cases where trailets are not appropriate (e.g. where some python object is
```
Comment (pllim) [inline docs/dev/ui_description.rst:271]: Are you sure this wasn't a pun...? `app`propriate? 😉

File: jdaviz/core/events.py [hunk 2]
```diff
@@ -51,14 +51,14 @@ def path(self):
...
+    def __init__(self, indices, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
-        self._indicies = indicies
+        self._indices = indices
 
     @property
-    def indicies(self):
-        return self._indicies
+    def indices(self):
```
Comment (pllim) [inline jdaviz/core/events.py:60]: This does actually change the API... @rosteen @javerbukh @ojustino , any objection on this change?

File: jdaviz/core/registries.py [hunk 3]
```diff
@@ -146,8 +146,8 @@ def decorator(cls):
 
 class MenuRegistry(UniqueDictRegistry):
     """
-    Registry containg referenecs to plugins that will populate the application-
-    level menu bar.
+    Registry containing referenecs to plugins that will populate the
```
Comment (pllim) [inline jdaviz/core/registries.py:149]: Missed one!
```suggestion
    Registry containing references to plugins that will populate the
```

File: jdaviz/core/events.py (hunk 2 above)
Comment (larrybradley) [reply jdaviz/core/events.py:60]: I didn't see where this is used anywhere.  And I assume that external users are using `DataSelectedMessage`?

Comment (larrybradley) [reply jdaviz/core/events.py:60]: Which begs the question -- is `DataSelectedMessage` even needed?

Comment (rosteen) [reply jdaviz/core/events.py:60]: This looks like an orphan message type that may have been used before, but can be removed now (as far as I can tell too).

Comment (pllim) [reply jdaviz/core/events.py:60]: 🤷

File: docs/dev/ui_description.rst (hunk 1 above)
Comment (rosteen) [reply docs/dev/ui_description.rst:271]: You may as well fix "trailets" to "traitlets" while you're at it 😆

Comment (rosteen) [reply docs/dev/ui_description.rst:271]: And now having looked at that doc page...it's actually wrong throughout the whole page! I had to check https://traitlets.readthedocs.io/en/stable/ and make sure I wasn't remembering the name wrong.

Comment (pllim) [reply docs/dev/ui_description.rst:271]: Is putting comma after `e.g.` out of scope? 👀

Comment (larrybradley) [reply docs/dev/ui_description.rst]: trailets -> traitlets is done.  There were several instances.

File: jdaviz/core/events.py (hunk 2 above)
Comment (larrybradley) [reply jdaviz/core/events.py:60]: Should I remove `DataSelectedMessage` in a separate PR or here?  It's a simple change, but not really a typo fix.

File: docs/dev/ui_description.rst (hunk 1 above)
Comment (larrybradley) [reply docs/dev/ui_description.rst]: commas added after `.e.g.` too (there were many other instances). :)

File: jdaviz/core/events.py (hunk 2 above)
Comment (pllim) [reply jdaviz/core/events.py:60]: Might as well just do it here? One less PR to review... 😬

File: jdaviz/core/registries.py (hunk 3 above)
Comment (pllim) [reply jdaviz/core/registries.py:149]: What about this one, @larrybradley ?

File: jdaviz/core/events.py (hunk 2 above)
Comment (larrybradley) [reply jdaviz/core/events.py:60]: There appears to be several unused `*Message` classes, so I opened a new PR:  https://github.com/spacetelescope/jdaviz/pull/873

**Review by pllim — APPROVED** [review]
OK no reason to delay. Thanks!

**Pull request merged**